src/tasks/components/TaskScheduler/ScriptSelector.tsx

```diff
@@ -1,44 +1,149 @@
   Dropdown,
+  EmptyState,
   Form,
   IconFont,
   Input,
+  RemoteDataState,
+  TechnoSpinner,
 } from '@influxdata/clockface'
 
+// APIs
+import {fetchScripts} from 'src/scripts/apis/index'
+
+// Notifications
+import {getScriptsFail} from 'src/shared/copy/notifications/categories/scripts'
+import {notify} from 'src/shared/actions/notifications'
+
+// Types
+import {Script} from 'src/types/scripts'
+
 // Utils
 import {SafeBlankLink} from 'src/utils/SafeBlankLink'
 
-export const ScriptSelector: FC = () => {
+interface Props {
+  selectedScript: Script
+  setSelectedScript: (script: Script) => void
+}
+
+export const ScriptSelector: FC<Props> = ({
+  selectedScript,
+  setSelectedScript,
+}) => {
+  const [scripts, setScripts] = useState<Script[]>([])
+  const [searchTerm, setSearchTerm] = useState('')
+  const [scriptsLoadingStatus, setScriptsLoadingStatus] = useState(
+    RemoteDataState.NotStarted
+  )
+  const dispatch = useDispatch()
   const {orgID} = useParams<{orgID: string}>()
   const history = useHistory()
+
   const fluxScriptEditorLink = `/orgs/${orgID}/data-explorer?fluxScriptEditor`
 
+  useEffect(() => {
+    setScriptsLoadingStatus(RemoteDataState.Loading)
+    const getScripts = async () => {
+      try {
+        const scripts = await fetchScripts()
+        setScripts(scripts.scripts)
+        setScriptsLoadingStatus(RemoteDataState.Done)
+      } catch (error) {
+        setScriptsLoadingStatus(RemoteDataState.Error)
+        dispatch(notify(getScriptsFail()))
+      }
+    }
+    getScripts()
+  }, [dispatch])
+
+  const filteredScripts = () =>
+    scripts.filter(script =>
+      script.name.toLocaleLowerCase().includes(searchTerm.toLocaleLowerCase())
+    )
+
+  const searchForTerm = (event: ChangeEvent<HTMLInputElement>) => {
+    setSearchTerm(event.target.value)
+  }
+
   const openScriptEditor = () => {
     history.push(`/orgs/${orgID}/data-explorer?fluxScriptEditor`)
   }
 
-  const createScriptInEditor = (
-    <Dropdown.Item onClick={openScriptEditor}>
-      + Create Script in Editor
-    </Dropdown.Item>
-  )
+  const handleSelectScript = script => {
+    setSelectedScript(script)
+  }
+
+  let scriptsList
 
-  const dropdownButtonText = 'Select a Script'
+  if (
+    scriptsLoadingStatus === RemoteDataState.NotStarted ||
+    scriptsLoadingStatus === RemoteDataState.Loading
+  ) {
+    scriptsList = (
+      <div>
+        <TechnoSpinner strokeWidth={ComponentSize.Small} diameterPixels={32} />
+      </div>
+    )
+  }
+
+  if (scriptsLoadingStatus === RemoteDataState.Error) {
+    scriptsList = (
+      <div>
+        <p>Could not get scripts</p>
+      </div>
+    )
+  }
+
+  if (scriptsLoadingStatus === RemoteDataState.Done) {
+    scriptsList = (
+      <>
+        {filteredScripts().map(script => (
+          <Dropdown.Item
+            key={script.id}
+            value={script.name}
+            onClick={() => handleSelectScript(script)}
+            selected={script.name === selectedScript?.name}
```

Comment:
I see what you're saying. Let me explain my thought process here - so, the `handleSelectScript` function is declared outside the onClick prop. The reason the anonymous function is needed here is - without it, `handleSelectScript()` will get called during render. and we don't want that. It should only get called when the button is clicked. Which is why wrapping it with an anonymous function is important in this specific case.

Reply:
I missed this conversation earlier.

So, looking into Clockface, the `Dropdown.Item` will pass the `value` prop to the `onClick` handler if you use the callback function's first argument. In other words
```
onClick={handleSelectScript}
```
would receive the `value` prop rather than the `script`.

So, i guess this works here.